lib/Backends/Interpreter/Interpreter.cpp

```diff
@@ -182,31 +184,42 @@ bool Interpreter::isOpSupported(const NodeInfo &NI) const {
                {SelectNode::CondIdx}) &&
            (NI.getInElemTy(SelectNode::CondIdx) == ElemKind::BoolTy);
 
-  case Kinded::Kind::CmpLTENodeKind:
-    return NI.allInputsAndOutputsHaveSameElemKind(
-               {ElemKind::FloatTy, ElemKind::Float16Ty, ElemKind::Int8QTy}, {},
-               {CmpLTENode::ResultIdx}) &&
-           (NI.getOutElemTy(CmpLTENode::ResultIdx) == ElemKind::BoolTy);
+  case Kinded::Kind::NotNodeKind:
+  case Kinded::Kind::AndNodeKind:
+  case Kinded::Kind::OrNodeKind:
+  case Kinded::Kind::XorNodeKind:
+    return NI.allInputsAndOutputsHaveSameElemKind({ElemKind::BoolTy});
+
+  case Kinded::Kind::AbsNodeKind:
+  case Kinded::Kind::NegNodeKind:
+  case Kinded::Kind::FloorNodeKind:
+  case Kinded::Kind::CeilNodeKind:
+  case Kinded::Kind::RoundNodeKind:
+  case Kinded::Kind::SqrtNodeKind:
+  case Kinded::Kind::RsqrtNodeKind:
+  case Kinded::Kind::ReciprocalNodeKind:
+  case Kinded::Kind::SinNodeKind:
+  case Kinded::Kind::CosNodeKind:
+    return NI.allInputsAndOutputsHaveSameElemKind(
+        {ElemKind::FloatTy, ElemKind::Int8QTy});
 
+  case Kinded::Kind::CmpEQNodeKind:
+  case Kinded::Kind::CmpNEQNodeKind:
   case Kinded::Kind::CmpLTNodeKind:
+  case Kinded::Kind::CmpLTENodeKind:
+  case Kinded::Kind::CmpGTNodeKind:
+  case Kinded::Kind::CmpGTENodeKind:
     return NI.allInputsAndOutputsHaveSameElemKind(
                {ElemKind::FloatTy, ElemKind::Float16Ty, ElemKind::Int8QTy,
                 ElemKind::Int32ITy, ElemKind::Int64ITy},
-               {}, {CmpLTNode::ResultIdx}) &&
-           (NI.getOutElemTy(CmpLTNode::ResultIdx) == ElemKind::BoolTy);
+               {}, {CmpEQNode::ResultIdx}) &&
```

Comment:
Well I don't think it is a problem because all the nodes are similar and they have only one output (ResultIdx = 0). It would make no sense for any of them to diverge from the rest.